### PR TITLE
Feat: pipeline API for apiserver

### PR DESCRIPTION
**Pull request by chivalryq**
Signed-off-by: Qiaozp <qiaozhongpei.qzp@alibaba-inc.com>### Description of your changes

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Review by FogDong — APPROVED** [review]

**Review by barnettZQG — APPROVED** [review]

File: pkg/apiserver/domain/model/context.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+// TableName return custom table name
+func (c *PipelineContext) TableName() string {
+	return tableNamePrefix + "pipeline_context"
+}
+
+// ShortTableName is the compressed version of table name for kubeapi storage and others
+func (c *PipelineContext) ShortTableName() string {
+	return "context"
```
Comment (wonderflow) [inline pkg/apiserver/domain/model/context.go:46]:
```suggestion
	return "pp_ctx"
```

File: pkg/apiserver/interfaces/api/dto/v1/types.go [hunk 2]
```diff
@@ -48,6 +48,12 @@ var (
 	CtxKeyUser = "user"
 	// CtxKeyToken request context key of request token
 	CtxKeyToken = "token"
+	// CtxKeyPipeline request context key of pipeline
+	CtxKeyPipeline = "pipeline"
+	// CtxKeyPipelineContex request context key of pipeline context
+	CtxKeyPipelineContex = "pipeline-context"
+	// CtxKeyPipelineRun request context key of pipeline run
+	CtxKeyPipelineRun = "pipelineRun"
```
Comment (wonderflow) [inline pkg/apiserver/interfaces/api/dto/v1/types.go:56]:
```suggestion
	CtxKeyPipelineRun = "pipeline-run"
```

align the name rule

File: pkg/apiserver/utils/bcode/016_pipeline.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package bcode
+
+var (
+	// ErrContextNotFound means the certain context is not found
+	ErrContextNotFound = NewBcode(400, 16001, "context is not found")
```
Comment (wonderflow) [inline pkg/apiserver/utils/bcode/016_pipeline.go:21]:
```suggestion
	ErrContextNotFound = NewBcode(400, 16001, "context of pipeline is not found")
```

File: pkg/apiserver/utils/bcode/016_pipeline.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+	// ErrContextAlreadyExist means the certain context already exists
+	ErrContextAlreadyExist = NewBcode(400, 16002, "context already exist")
```
Comment (wonderflow) [inline pkg/apiserver/utils/bcode/016_pipeline.go:23]:
```suggestion
	ErrContextAlreadyExist = NewBcode(400, 16002, "pipeline context already exist")
```

**Review by wonderflow — APPROVED** [review]

**Pull request merged**